### ISSUE TITLE
Add a Cache.deleteStudent() method instead of using Cache.setStudent(null, null)

### ISSUE THREAD

**Issue by abhijitparida** [abhijitparida/bunk#21]
File location: app/src/main/java/app/abhijit/iter/data/Cache.java

(update/add unit tests accordingly)

**Comment by jashasweejena** [comment]
I'd like to take this up!

**Comment by abhijitparida** [comment]
Alright! 🎉

**Comment by jashasweejena** [comment]
Do I need to modify only the implementation in **data/Cache.java** or do I need to modify the method uses too?

**Comment by abhijitparida** [comment]
You also need to change all the method calls.
Bonus: add unit tests for the new method.

**Comment by jashasweejena** [comment]
> You also need to change all the method calls.
> Bonus: add unit tests for the new method.

Never added a unit test myself. Could you walk me through briefly, if you don't mind? 😃

**Comment by jashasweejena** [comment]
Could I work on it at the night? if it's all good with you!

**Comment by abhijitparida** [comment]
I fixed the build config issue. Update your fork accordingly. I also recommend using different branches for each pull request. See: https://guides.github.com/introduction/flow/

> Never added a unit test myself. Could you walk me through briefly, if you don't mind? 😃
Actually, I haven't set up tests for `Cache` yet. So you don't have to worry about that.

Here is an example unit test, for `Subject`: [SubjectTest.java](https://github.com/abhijitparida/bunk/blob/master/app/src/test/java/app/abhijit/iter/models/SubjectTest.java)

> Could I work on it at the night? if it's all good with you!
Take your time!

**Comment by jashasweejena** [comment]
> Actually, I haven't set up tests for `Cache` yet. So you don't have to worry about that.

So I'd have to add tests for the entire 'Cache.java' if I'm not mistaken?
> Here is an example unit test, for `Subject`: [SubjectTest.java](https://github.com/abhijitparida/bunk/blob/master/app/src/test/java/app/abhijit/iter/models/SubjectTest.java)

Thank you!
> 

> Take your time!

Sure 😁

**Comment by abhijitparida** [comment]
No, you don't have to add tests for `Cache`.

**Comment by jashasweejena** [comment]
Okay. Thank you!

**Comment by jashasweejena** [comment]
> No, you don't have to add tests for `Cache`.

So, all in all, I don't need to work with tests?

**Comment by abhijitparida** [comment]
No. Just ensure that the app builds successfully without any warnings/errors.

**Comment by pedroqueiroz** [comment]
Hi! 

Is this issue taken by anyone?

Thanks! :)
